server/bin/pbench-state-manager.py

```diff
@@ -49,15 +50,27 @@ def main(options):
         init_db(config, logger)
 
         args = {}
+
+        if not (options.path or options.md5):
+            print(
+                f"{_NAME_}: Either --path or --md5 must be specified",
+                file=sys.stderr,
+            )
+            return 2
+
         if options.create:
             args["owner"] = options.create
 
         if options.path:
-            args["name"] = Dataset.stem(options.path)
-        elif options.name:
+            if not options.md5:
+                args["resource_id"] = md5sum(options.path).md5_hash
+            if options.create and not options.name:
+                args["name"] = Dataset.stem(options.path)
+        elif options.name and options.create:
             args["name"] = options.name
         if options.md5:
-            args["md5"] = options.md5
+            args["resource_id"] = options.md5
```

Comment:
Alternately,
```
        if options.md5:
            args["resource_id"] = options.md5
        elif options.path:
            args["resource_id"] = md5sum(options.path).md5_hash

        if options.create:
            if options.name:
                args["name"] = options.name
            elif options.path:
                args["name"] = Dataset.stem(options.path)
            else:
                I_think_we_need_to_fail_here()
```
I think this approach is simpler, and it addresses the case when `create`, `path`, and `name` are all true-y, which I think your code misses.  (I don't know whether we need/want to handle the failure at the last line, but it seems like a good idea and that seems like a good place to do it.)